Decred Wallet/Features/Receive/ReceiveViewController.swift

```diff
@@ -0,0 +1,219 @@
+//  ReceiveViewController.swift
+//  Decred Wallet
+//
+// Copyright (c) 2018-2020 The Decred developers
+// Use of this source code is governed by an ISC
+// license that can be found in the LICENSE file.
+
+import Foundation
+import UIKit
+import Dcrlibwallet
+
+class ReceiveViewController: UIViewController, UIDocumentInteractionControllerDelegate {
+    @IBOutlet weak var menuBtn: UIButton!
+    @IBOutlet private var imgWalletAddrQRCode: UIImageView!
+
+    @IBOutlet weak var walletAddressLabel: UILabel!
+    @IBOutlet weak var containerRoundedView: RoundedView!
+
+    @IBOutlet weak var selectedAccountView: UIView!
+    @IBOutlet weak var accountNameLabel: UILabel!
+    @IBOutlet weak var walletNameLabel: UILabel!
+    @IBOutlet weak var totalAccountBalanceLabel: UILabel!
+
+    private lazy var syncInProgressLabel: UILabel = {
+        let label = UILabel()
+        label.translatesAutoresizingMaskIntoConstraints = false
+        label.textColor = .darkGray
+        label.numberOfLines = 0
+        label.textAlignment = .center
+        label.text = LocalizedStrings.secureMenuSyncInfo
+        return label
+    }()
+
+    var tapGesture = UITapGestureRecognizer()
+    var oldAddress = ""
+
+    var selectedWallet: Wallet?
+    var selectedAccount: DcrlibwalletAccount?
+
+    override func loadView() {
+        super.loadView()
+        view.addSubview(syncInProgressLabel)
+    }
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        // TAP Gesture
+        self.setupExtraUI()
+        setupSyncInProgressLabelConstraints()
+    }
+
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+        self.checkSyncStatus()
+    }
+
+    func setupExtraUI() {
+        self.imgWalletAddrQRCode.addGestureRecognizer(UITapGestureRecognizer(target: self, action: #selector(self.copyAddress)))
+        self.walletAddressLabel.addGestureRecognizer(UITapGestureRecognizer(target: self, action: #selector(self.copyAddress)))
+        self.selectedAccountView.addGestureRecognizer(UITapGestureRecognizer(target: self, action: #selector(self.showAccountSelectDialog)))
+    }
+
+    @objc func copyAddress() {
+        DispatchQueue.main.async {
+            //Copy a string to the pasteboard.
+            UIPasteboard.general.string = self.walletAddressLabel.text!
+            Utils.showBanner(parentVC: self, type: .success, text: LocalizedStrings.walletAddrCopied)
+        }
+    }
+
+    private func checkSyncStatus() {
+        self.menuBtn.isEnabled = false
+    
+        if let wallet = WalletLoader.shared.wallets.map({ Wallet.init($0) }).first,
```

Comment:
```suggestion
        if let wallet = WalletLoader.shared.wallets.filter({ $0.hasDiscoveredAccounts || !$0.isRestored }).first,
```

Use the first wallet that has either completed account discovery or wasn't restored in the first place. I don't even think it is possible for a wallet that wasn't restored to have `hasDiscoveredAccounts == false`, but I'll need to confirm.